File: scripts/hooks/command-audit-log.js
#!/usr/bin/env node
/**
 * command-audit-log.js — Audit log bash commands
 *
 * Logs all bash commands with timestamps for session review.
 * Writes to .opc/audit.log (or falls back to console).
 */

const fs = require('fs');
const path = require('path');

try {
  const raw = fs.readFileSync(0, 'utf8');
  const input = JSON.parse(raw);
  const command = (input.tool_input && input.tool_input.command) || '';

  if (!command) process.exit(0);

  const timestamp = new Date().toISOString();
  const logLine = `[${timestamp}] ${command}\n`;

  // Try to write to .opc/audit.log
  const logDir = path.join(process.cwd(), '.opc');
  const logFile = path.join(logDir, 'audit.log');

  try {
    if (!fs.existsSync(logDir)) {
      fs.mkdirSync(logDir, { recursive: true });
    }
    fs.appendFileSync(logFile, logLine);
  } catch (writeErr) {
    // Can't write log file — not critical, skip silently
  }
} catch (e) {
  // Silent pass
}

process.exit(0);

File: scripts/hooks/command_audit_log.py
#!/usr/bin/env python3

from __future__ import annotations

from datetime import datetime, timezone
from pathlib import Path
import sys

from common import ensure_dir, get_tool_input, read_stdin_json


def main() -> int:
    payload = read_stdin_json()
    command = get_tool_input(payload).get("command", "")
    if not isinstance(command, str) or not command:
        return 0

    timestamp = datetime.now(timezone.utc).isoformat()
    log_dir = Path.cwd() / ".opc"
    log_file = log_dir / "audit.log"

    try:
        ensure_dir(log_dir)
        with log_file.open("a", encoding="utf-8") as handle:
            handle.write(f"[{timestamp}] {command}\n")
    except OSError:
        pass

    return 0


if __name__ == "__main__":
    sys.exit(main())
